parlai/core/torch_generator_agent.py

```diff
@@ -942,12 +947,13 @@ def _get_rescored_finished(self, n_best=None):
         """
         # if we never actually finished, force one
         if not self.finished:
+            self.outputs[-1][0] = self.eos
             self.finished.append(
                 _HypothesisTail(
                     timestep=len(self.outputs) - 1,
                     hypid=0,
                     score=self.all_scores[-1][0],
-                    tokenid=self.eos,
+                    tokenid=self.outputs[-1][0],
```

Comment:
Do we need these changes?

Reply:
basically we can keep self.eos, but using outputs (previously updated) makes a clear reason why we have updated it, so I would keep it like this